pkg/controller/hpa/advanced_hpa_controller.go

```diff
@@ -43,25 +46,92 @@ func (p *AdvancedHPAController) Reconcile(ctx context.Context, req ctrl.Request)
+			p.UpdateStatus(ctx, ahpa, newStatus)
 			return ctrl.Result{}, err
 		}
-	}
 
-	err = p.ReconcileHPA(ctx, ahpa)
-	if err != nil {
-		return ctrl.Result{}, err
+		newStatus.ExpectReplicas = ahpa.Spec.SpecificReplicas
+		newStatus.CurrentReplicas = &scale.Status.Replicas
+
+		// todo: validation SpecificReplicas is between minReplicas and maxReplicas in webhook
+		// scale target to its specific replicas
+		if ahpa.Spec.SpecificReplicas != nil && *ahpa.Spec.SpecificReplicas != scale.Status.Replicas {
+			scale.Spec.Replicas = *ahpa.Spec.SpecificReplicas
+			updatedScale, err := p.scaleClient.Scales(scale.Namespace).Update(ctx, mapping.Resource.GroupResource(), scale, metav1.UpdateOptions{})
+			if err != nil {
+				p.Recorder.Event(ahpa, v1.EventTypeNormal, "FailedManualScale", err.Error())
+				p.Log.Error(err, "Failed to manual scale target to specific replicas", "advanced-hpa", klog.KObj(ahpa), "replicas", ahpa.Spec.SpecificReplicas)
+				setCondition(newStatus, autoscalingapi.Ready, metav1.ConditionFalse, "FailedScale", "Failed to scale target manually")
+				p.UpdateStatus(ctx, ahpa, newStatus)
+				return ctrl.Result{}, err
+			}
+
+			now := metav1.Now()
+			newStatus.LastScaleTime = &now
+			newStatus.CurrentReplicas = &updatedScale.Status.Replicas
+		}
+	} else if ahpa.Spec.ScaleStrategy == autoscalingapi.ScaleStrategyAuto {
+		// reconcile prediction if enabled
+		if IsPredictionEnabled(ahpa) {
+			prediction, err := p.ReconcilePodPredication(ctx, ahpa)
+			if err != nil {
+				setCondition(newStatus, autoscalingapi.Ready, metav1.ConditionFalse, "FailedReconcilePrediction", err.Error())
+				p.UpdateStatus(ctx, ahpa, newStatus)
+				return ctrl.Result{}, err
+			}
+			setPredictionCondition(newStatus, prediction.Status.Conditions)
+		}
+
+		hpa, err := p.ReconcileHPA(ctx, ahpa)
+		if err != nil {
+			setCondition(newStatus, autoscalingapi.Ready, metav1.ConditionFalse, "FailedReconcileHPA", err.Error())
+			p.UpdateStatus(ctx, ahpa, newStatus)
+			return ctrl.Result{}, err
+		}
+
+		newStatus.ExpectReplicas = &hpa.Status.DesiredReplicas
+		newStatus.LastScaleTime = hpa.Status.LastScaleTime
+		newStatus.CurrentReplicas = &hpa.Status.CurrentReplicas
+
+		setHPACondition(newStatus, hpa.Status.Conditions)
 	}
 
-	// todo: update status
-	// autoscalingStatusOrigin := autoscaler.Status.DeepCopy()
+	setCondition(newStatus, autoscalingapi.Ready, metav1.ConditionTrue, "AdvancedHorizontalPodAutoscalerReady", "Advanced HPA is ready")
+	return ctrl.Result{}, p.UpdateStatus(ctx, ahpa, newStatus)
+}
+
+func (p *AdvancedHPAController) UpdateStatus(ctx context.Context, ahpa *autoscalingapi.AdvancedHorizontalPodAutoscaler, newStatus *autoscalingapi.AdvancedHorizontalPodAutoscalerStatus) error {
+	if !equality.Semantic.DeepEqual(&ahpa.Status, newStatus) {
+		p.Log.Info("AdvancedHorizontalPodAutoscaler status should be updated", "currentStatus", &ahpa.Status, "newStatus", newStatus)
```

Comment:
can we always set log level for info?
and if you are dumping the entire status, the level should be at least 4.